rpcs3/Emu/RSX/VK/VKHelpers.h

```diff
@@ -2698,8 +2750,12 @@ namespace vk
 					createInfo.window                     = p.second;
 					CHECK_RESULT(vkCreateXlibSurfaceKHR(this->m_instance, &createInfo, nullptr, &m_surface));
 				}
+#endif
+#if defined(HAVE_X11) && defined(VK_USE_PLATFORM_WAYLAND_KHR)
+				else
```

Comment:
This check can be moved either above or below. The code is hard to follow like this.